client/client.go

```diff
@@ -180,17 +201,18 @@ func (client *Client) readLoop() {
 
 		contents := make([]byte, length)
 
-		if _, err = io.ReadFull(client.rw, contents); err != nil {
+		if _, err = io.ReadFull(client.conn, contents); err != nil {
 			if err = client.reconnect(err); err != nil {
 				break
 			}
 
 			continue
 		}
 
-		resp = getResponse()
+		resp = client.responsePool.Get().(*Response)
 
 		resp.DataType, err = rt.NewPT(pt)
+
 		if err != nil {
 			client.err(err)
 			continue
```

Comment:
This continue does nothing, right? I know it isn't your code, but trying to figure out golang.

Reply:
It returns to the beginning of the for loop bypassing the below switch statement. Actually now that I'm looking at this; this code should be returning the `resp` object to the pool.

Tsk tsk...